### PR TITLE
Remove Submenu block

### PR DESCRIPTION
**Pull request by mardelnet**
**DESCRIPTION:**
As the Submenu block was moved to the master theme, this PR removes it completely from the plugin repository.

**RELATED PRs:**
- https://github.com/greenpeace/planet4-master-theme/pull/2376
- https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/1239

### CODE REVIEW

File: planet4-gutenberg-blocks.php [hunk 1]
```diff
@@ -134,7 +134,6 @@
 	'planet4-blocks/media-video',
 	'planet4-blocks/social-media',
 	'planet4-blocks/spreadsheet',
-	'planet4-blocks/submenu',
```
Comment (mleray) [inline planet4-gutenberg-blocks.php:137]: We do need to keep these here for now, to make sure we select which blocks are allowed per post type. There is a ticket to move this filter to the theme.

Comment (mleray) [reply planet4-gutenberg-blocks.php:137]: Ah it's actually your [ticket](https://jira.greenpeace.org/browse/PLANET-7615) haha

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [nix](https://www-dev.greenpeace.org/test-nix) | [admin](https://www-dev.greenpeace.org/test-nix/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-nix/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-nix) | [composer-local.json](https://github.com/greenpeace/planet4-test-nix/blob/main/composer-local.json)

:watch: 2024.09.12 08:54:50

File: planet4-gutenberg-blocks.php (hunk 1 above)
Comment (mardelnet) [reply planet4-gutenberg-blocks.php:137]: Let's start by working-reviewing-merging [that ticket](https://jira.greenpeace.org/browse/PLANET-7615), then.

Comment (mleray) [reply planet4-gutenberg-blocks.php:137]: Why? When moving this filter to the theme it will also need to keep the Submenu block in it

Comment (mleray) [reply planet4-gutenberg-blocks.php:137]: So if you just keep the submenu block in this filter, we can merge the PR already, the rest looks good to me 👍

**Review by mleray — COMMENTED** [review]
I see we still have some [Submenu-related images](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/tree/main/public/images), that are no longer used, even in the theme. We can remove them too!

**Review by mleray — APPROVED** [review]
Looks good to me 👍

**Pull request merged**